src/test/kotlin/no/nav/tilleggsstonader/sak/vedtak/OpphørValidatorServiceTest.kt

```diff
@@ -0,0 +1,128 @@
+        }
+        assertThat(feil.message).isEqualTo("Det er vilkår eller vilkårperiode med vilkårstatus NY og resultat OPPFYLT.")
+    }
+
+    @Test
+    fun kasterFeilVedIngenNyeOppfylteAktivteter() {
+        every { vilkårService.hentVilkår(saksbehandling.id) } returns listOf(vilkår(behandlingId = saksbehandling.id, type = VilkårType.PASS_BARN, resultat = Vilkårsresultat.OPPFYLT, status = VilkårStatus.UENDRET))
+        every { vilkårperiodeService.hentVilkårperioder(saksbehandling.id) } returns Vilkårperioder(emptyList(), listOf(VilkårperiodeTestUtil.aktivitet()))
+        every { tilsynBarnBeregningService.beregn(saksbehandling) } returns BeregningsresultatTilsynBarn(emptyList())
+
+        val feil: ApiFeil = assertThrows {
+            opphørValidatorService.validerOpphør(saksbehandling)
+        }
+        assertThat(feil.message).isEqualTo("Det er vilkår eller vilkårperiode med vilkårstatus NY og resultat OPPFYLT.")
+    }
+
+    @Test
+    fun kasterFeilVedUtbetalingEtterOpphør() {
+        every { vilkårService.hentVilkår(saksbehandling.id) } returns listOf(vilkår(behandlingId = saksbehandling.id, type = VilkårType.PASS_BARN, resultat = Vilkårsresultat.OPPFYLT, status = VilkårStatus.UENDRET))
+        every { vilkårperiodeService.hentVilkårperioder(saksbehandling.id) } returns Vilkårperioder(emptyList(), emptyList())
+        every { tilsynBarnBeregningService.beregn(saksbehandling) } returns vedtakBeregningsresultat
+
+        val feil: ApiFeil = assertThrows {
+            opphørValidatorService.validerOpphør(saksbehandling)
+        }
+        assertThat(feil.message).isEqualTo("Det er utbetalinger etter opphørsdato")
+    }
+
+    @Test
+    fun kasterFeilVedMålgruppePeriodeFlyttetTilEtterOpphørtDato() {
+        every { vilkårService.hentVilkår(saksbehandling.id) } returns listOf(vilkår(behandlingId = saksbehandling.id, type = VilkårType.PASS_BARN, resultat = Vilkårsresultat.OPPFYLT, status = VilkårStatus.UENDRET))
+        every { vilkårperiodeService.hentVilkårperioder(saksbehandling.id) } returns Vilkårperioder(listOf(VilkårperiodeTestUtil.målgruppe(status = Vilkårstatus.ENDRET)), emptyList())
+        every { tilsynBarnBeregningService.beregn(saksbehandling) } returns BeregningsresultatTilsynBarn(emptyList())
+
+        val feil: ApiFeil = assertThrows {
+            opphørValidatorService.validerOpphør(saksbehandling)
+        }
+        assertThat(feil.message).isEqualTo("TOM er etter opphørsdato for endret målgruppe")
+    }
+
+    @Test
+    fun kasterFeilVedAktivitetPeriodeFlyttetTilEtterOpphørtDato() {
+        every { vilkårService.hentVilkår(saksbehandling.id) } returns listOf(vilkår(behandlingId = saksbehandling.id, type = VilkårType.PASS_BARN, resultat = Vilkårsresultat.OPPFYLT, status = VilkårStatus.UENDRET))
+        every { vilkårperiodeService.hentVilkårperioder(saksbehandling.id) } returns Vilkårperioder(emptyList(), listOf(VilkårperiodeTestUtil.aktivitet(status = Vilkårstatus.ENDRET)))
```

Comment:
Burde man teste med både resultat oppfylt og ikke oppfylt?

Reply:
Man skal uansett ikke få endre en periode til å ha tom etter opphørsdato, så siden koden ikke oppfører seg forskjellig ved de to casene trenger vi kanskje heller ikke å teste det?